File: src/test/scala/com/martomate/tripaint/ExtraAsserts.scala
package com.martomate.tripaint

import munit.Assertions.assertEquals

object ExtraAsserts {
  extension [T](s: Seq[T]) private def occurrences = s.groupMapReduce(identity)(_ => 1)(_ + _)

  def assertSameElementsIgnoringOrder[T](left: Seq[T], right: Seq[T])(using munit.Location): Unit =
    assertEquals(left.occurrences, right.occurrences)
}

File: core/src/test/kotlin/tripaint/ExtraAsserts.kt
package tripaint

import org.junit.jupiter.api.Assertions.assertEquals

object ExtraAsserts {
    private fun <T> List<T>.occurrences(): Map<T, Int> {
        val m = mutableMapOf<T, Int>()
        for (elem in this) {
            val k = elem
            val b = m[k]
            val v = if (b != null) b + 1 else 1
            m[k] = v
        }
        return m.toMap()
    }

    fun <T> assertSameElementsIgnoringOrder(left: List<T>, right: List<T>) = assertEquals(left.occurrences(), right.occurrences())
}